sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java

```diff
@@ -187,7 +189,11 @@ static String getDefaultScopeFromUrl(String url, AzureEnvironment environment) {
                     resource = removeTrailingSlash(resource);
                     break;
                 } else if (endpoint.getKey().equals(AzureEnvironment.Endpoint.STORAGE.identifier())) {
-                    resource = "https://storage.azure.com";
+                    try {
+                        resource = String.format("https://%s/", new URL(url).getAuthority());
+                    } catch (MalformedURLException e) {
+                        resource = "https://storage.azure.com";
```

Comment:
Do we have a diff of one trailing `/` in this 2 pattern?

Reply:
Also, may just paste the link you have (https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-access-azure-active-directory#microsoft-authentication-library-msal), as comment to this code block.

Reply:
updated